src/test/java/net/snowflake/client/jdbc/ProxyLatestIT.java

```diff
@@ -0,0 +1,405 @@
+    addProxyProperties(props);
+    props.put("nonProxyHosts", "notMatchingHost");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  @Test
+  public void testProxyNotUsedWhenNonProxyHostsMatchingInJVMParams() throws SQLException {
+    restoreTrustStorePathProperty();
+    Properties props = getProperties();
+    setJvmProperties(props);
+    System.setProperty("http.nonProxyHosts", "*");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyNotUsed();
+  }
+
+  @Test
+  public void testProxyUsedWhenNonProxyHostsNotMatchingInJVMParams() throws SQLException {
+    Properties props = getProperties();
+    setJvmProperties(props);
+    System.setProperty("http.nonProxyHosts", "notMatchingHost");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  private void startWiremockStandAlone() {
+    // retrying in case of fail in port bindings
+    await()
+        .alias("wait for wiremock responding")
+        .atMost(Duration.ofSeconds(20))
+        .until(
+            () -> {
+              try {
+                httpProxyPort = findFreePort();
+                httpsProxyPort = findFreePort();
+                wiremockStandalone =
+                    new ProcessBuilder(
+                            "java",
+                            "-jar",
+                            getWiremockStandAlonePath(),
+                            "--root-dir",
+                            System.getProperty("user.dir")
+                                + File.separator
+                                + WIREMOCK_HOME_DIR
+                                + File.separator,
+                            "--enable-browser-proxying", // work as forward proxy
+                            "--proxy-pass-through",
+                            "false", // pass through only matched requests
+                            "--port",
+                            String.valueOf(httpProxyPort),
+                            "--https-port",
+                            String.valueOf(httpsProxyPort),
+                            "--https-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"),
+                            "--ca-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"))
+                        .inheritIO()
+                        .start();
+                waitForWiremock();
+                return true;
+              } catch (Exception e) {
+                logger.info(
+                    "Failed to start wiremock, retrying: " + Arrays.toString(e.getStackTrace()));
+                return false;
+              }
+            });
+  }
+
+  private static void downloadWiremock() {
+    try (CloseableHttpClient client = HttpClients.createDefault();
+        CloseableHttpResponse response = client.execute(new HttpGet(WIREMOCK_STANDALONE_URL))) {
```

Comment:
@sfc-gh-mhofman @sfc-gh-pmotacki do I need to perform security review for wiremock-standalone ?

Reply:
yes

Reply:
created: https://github.com/snowflakedb/security_reviews/pull/5438